### PR TITLE
merging mcgill devel with very few changes

### PR DESCRIPTION
**Pull request by josuegbl**

### CODE REVIEW

**Comment by pconesa** [comment]
Thanks @josuegbl we are a bit busy preparing next week meeting and course....I looked at it and doesn't seem hard...I'll try to find some time to check it out and test.

**Comment by pconesa** [comment]
Hi @josuegbl 
We can't merge any PR that will affect xmipp code (c++ nor python)....until Xmipp split is done.

Is it possible to remove the 2 xmipp files involved here?

**Comment by josuegbl** [comment]
HI @pconesa, the only real change here is to solve a bug in xmipp (python) code. You can close it.

File: pyworkflow/em/packages/xmipp3/protocol_volume_homogenizer.py [hunk 1]
```diff
@@ -405,12 +406,12 @@ def _citations(self):
         return ['**********????????????????????************']
     
     def _setHalf1(self, item, row):
-        if item._rlnRandomSubset == 1:
-            item._appendItem = False
+        if (item._rln_halfId == 1):
```
Comment (delarosatrevin) [inline pyworkflow/em/packages/xmipp3/protocol_volume_homogenizer.py]: Parenthesis are not needed, also _rln_halfId does not complain our naming standard, not _ in the middle of the name. Moreover, if it is a Relion label, I think _rlnRandomSubset is more appropriate.

File: pyworkflow/em/packages/xmipp3/protocol_volume_homogenizer.py [hunk 2]
```diff
@@ -405,12 +406,12 @@ def _citations(self):
...
+            item._appendItem=False
```
Comment (delarosatrevin) [inline pyworkflow/em/packages/xmipp3/protocol_volume_homogenizer.py]: You removed the spaces around =, making the code non PEP8

File: pyworkflow/em/packages/xmipp3/protocol_volume_homogenizer.py [hunk 3]
```diff
@@ -405,12 +406,12 @@ def _citations(self):
...
 
     def _setHalf2(self, item, row):
-        if item._rlnRandomSubset == 2:
-            item._appendItem = False
+        if (item._rln_halfId == 2):
```
Comment (delarosatrevin) [inline pyworkflow/em/packages/xmipp3/protocol_volume_homogenizer.py]: Same as the previous comments.

**Comment by josuegbl** [comment]
@delarosatrevin, your suggestions has been addressed. Please, consider to merge it :smile:

**Pull request merged**